src/main/java/mod/gate/features/commands/ChangeSetting.java

```diff
@@ -0,0 +1,132 @@
+import mod.gate.core.command.Command;
+import mod.gate.core.config.ConfigHolder;
+import mod.gate.core.config.ConfigManager;
+import mod.gate.utils.ChatUtils;
+import net.fabricmc.fabric.api.client.command.v1.ClientCommandManager;
+import net.fabricmc.fabric.api.client.command.v1.FabricClientCommandSource;
+
+import java.lang.reflect.Field;
+import java.util.Objects;
+
+public class ChangeSetting implements Command {
+
+    @Override
+    public LiteralArgumentBuilder<FabricClientCommandSource> register() {
+        LiteralArgumentBuilder<FabricClientCommandSource> literalArgumentBuilder = ClientCommandManager.literal("gate");
+
+        for (ConfigHolder config : ConfigManager.getConfigs()) {
+
+            if (config.getField().getType().getName() == config.getField().getType().getSimpleName()) {
+                //primitive type
+                literalArgumentBuilder.then(ClientCommandManager.literal("changesetting").then(
+                        ClientCommandManager.literal(config.getField().getName())
+                                .then(ClientCommandManager.argument(
+                                                "value", StringArgumentType.greedyString())
+                                        .executes(this::run))
+                ));
+
+            } else {
+                //Object type
+                for (Field field: config.getField().getType().getFields()) {
+                    literalArgumentBuilder.then(ClientCommandManager.literal("changesetting").then(
+                            ClientCommandManager.literal(config.getField().getName())
+                                    .then(ClientCommandManager.literal(field.getName())
+                                    .then(ClientCommandManager.argument(
+                                                    "value", StringArgumentType.greedyString())
+                                            .executes(this::run)))
+                    ));
+                }
+            }
+        }
+
+        return (literalArgumentBuilder);
+    }
+
+    public int run(CommandContext<FabricClientCommandSource> context) {
+        //0 and 1 are commands (gate changesetting) 2 is setting name and 3 is either path or value and 4 is null or value
+        String[] args = context.getInput().split(" ");
+
+        for (ConfigHolder config: ConfigManager.getConfigs()) {
+            //skip if isn't the same
+            if (!config.getField().getName().equals(args[2])) continue;
+
+
+            try {
+                if (args.length > 4) {
+                    //has path and value
+
+                    for (Field field : config.getField().getType().getFields()) {
+                        if (!field.getName().equals(args[3])) continue;
+                        field.set(config.getField().get(config.getParent()), parsePrimitive(args[4], field.getType().getName()));
+
+                        config.setConfig(config.getField().get(config.getParent()));
```

Comment:
I wasn't sure if this was required and I tested it out and it isn't required, so if there are modifications to do I'll push it on next commit or please next person to do a pull request do it